index.js

```diff
@@ -70,20 +71,111 @@ function format(text, opts, addAlignmentSize) {
   return str;
 }
 
-function formatRange(text, opts) {
+function findNodeByOffset(node, offset) {
+  const start = util.locStart(node);
+  const end = util.locEnd(node);
+  if (start <= offset && offset <= end) {
+    for (const childNode of comments.getSortedChildNodes(node)) {
+      const childResult = findNodeByOffset(childNode, offset);
+      if (childResult) {
+        return childResult;
+      }
+    }
+
+    if (isSourceElement(node)) {
+      return node;
+    }
+  }
+}
+
+// See https://www.ecma-international.org/ecma-262/5.1/#sec-A.5
+function isSourceElement(node) {
+  if (node == null) {
+    return false;
+  }
+  switch (node.type) {
+    case "FunctionDeclaration":
+      return true;
+    // Statements
```

Comment:
You also have `VariableDeclaration` at the bottom. I would just remove the `return true; and // Statements` comment.